rust/src/coordinator.rs

```diff
@@ -230,31 +227,24 @@ impl Coordinator {
         .ok_or(PetError::InvalidMessage)
     }
 
-    /// Freeze the seed dictionary. Fails due to insufficient update participants.
-    fn freeze_dict_seed(&mut self) -> Result<(), PetError> {
-        (self
-            .dict_seed
-            .values()
-            .all(|dict| dict.len() >= self.min_update))
-        .then_some(())
-        .ok_or(PetError::InsufficientParticipants)
-    }
-
     /// Update the mask dictionary.
-    fn update_dict_mask(&mut self, mask_url: &[u8]) {
-        self.dict_mask += mask_url
-            .chunks_exact(mask_url.len())
-            .map(|mask| mask.to_vec());
+    fn update_dict_mask(&mut self, mask_hash: &[u8]) {
+        self.dict_mask.update(iter::once(mask_hash.to_vec()))
     }
 
-    /// Freeze the mask dictionary. Returns a unique mask. Fails due to insufficient sum
-    /// participants.
+    /// Freeze the mask dictionary.
     fn freeze_dict_mask(&self) -> Result<Vec<u8>, PetError> {
         let counts = self.dict_mask.most_common();
-        (counts.iter().map(|(_, count)| count).sum::<usize>() >= self.min_sum
-            && (counts.len() == 1 || counts[0].1 > counts[1].1))
-            .then(|| counts[0].0.clone())
-            .ok_or(PetError::InsufficientParticipants)
+
+        // it is an error to call freeze_dict_mask if we didn't
+        // receive a single mask
+        assert!(!counts.is_empty());
```

Comment:
this is already covered by `has_enough_masks()`, the `mask_count` yields `0` in this case.

Reply:
Yeah that's why I made is an `assert!`. But you're right, maybe it's useless I'll remove it.